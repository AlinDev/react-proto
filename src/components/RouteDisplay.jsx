import React from 'react';
import IconButton from '@material-ui/core/IconButton';
import DeleteIcon from '@material-ui/icons/Delete';
import PropTypes from 'prop-types';

const RouteDisplay = ({
  componentTitle,
  pathName,
  routeCompId,
  deleteRoute,
  classes,
  routerCompId,
}) => (
  <div className="routeListItem">
    Path: / {pathName}
    Component: {componentTitle}
    <IconButton
      className={classes.button}
      onClick={() => {
        deleteRoute({
          routeCompId,
          routerCompId,
        });
      }}
      aria-label='Delete'>
      <DeleteIcon className={classes.light} />
    </IconButton>
  </div>
);

export default RouteDisplay;

RouteDisplay.propTypes = {
  deleteRoute: PropTypes.func.isRequired,
  routerCompId: PropTypes.string.isRequired,
  componentTitle: PropTypes.string.isRequired,
  pathName: PropTypes.string.isRequired,
  routeCompId: PropTypes.string.isRequired,
  classes: PropTypes.object.isRequired,
};
