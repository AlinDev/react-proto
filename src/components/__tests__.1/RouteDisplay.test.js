import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import RouteDisplay from '../RouteDisplay.jsx';

describe('React unit tests', () => {
  describe('RouteDisplay', () => {
    let wrapper;
    const handleDeleteRoute = jest.fn();
    const classes = {};

    const props = {
      componentTitle: 'TestComp',
      pathName: 'TestPath',
      routeCompId: '2',
      handleDeleteRoute,
      classes,
      routerCompId: '1',
    };

    beforeAll(() => {
      wrapper = mount(<RouteDisplay {...props} />);
    });

    it('snapshot test', () => {
      const tree = renderer.create(<RouteDisplay {...props} />).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('Should show the path name and comp name, with one line break', () => {
      expect(wrapper.text()).toEqual('Path: / TestPathComponent: TestComp');
      expect(wrapper.find('br')).toHaveLength(1);
    });

    it('clicks delete handler when button clicked', () => {
      wrapper.find('button').simulate('click');
      expect(handleDeleteRoute.mock.calls.length).toEqual(1);
    });
  });
});
