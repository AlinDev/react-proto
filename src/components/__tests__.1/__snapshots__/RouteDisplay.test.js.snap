// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React unit tests RouteDisplay snapshot test 1`] = `
<div
  align="stretch"
  className="MuiGrid-container-1 MuiGrid-align-items-xs-baseline-12"
>
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-10-39"
  >
    Path: / 
    TestPath
    <br />
    Component: 
    TestComp
  </div>
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-2-31"
  >
    <button
      aria-label="Delete"
      className="MuiButtonBase-root-104 MuiIconButton-root-98"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiIconButton-label-103"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-107"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
          />
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
        </svg>
      </span>
      <span
        className="MuiTouchRipple-root-114"
      />
    </button>
  </div>
</div>
`;
